### ISSUE TITLE
Error being thrown by iconv, unable to catch it..

### ISSUE THREAD

**Issue by makecontact** [bitinn/node-fetch#410]
Hi,

Can't seem to catch this error but it's coming from inside node-fetch.  How to catch this / or prevent it? I've installed iconv (as hinted on another old forum) but that's not resolved it...

```
fetch(url, options).then(function(res) {
}).then(function(body) {
}).catch (function(err) {

//Not being caught here

});
```

The error isn't killing my app, but would prefer to handle it :)

`Error: Conversion from "UTF-8" to UTF-8//TRANSLIT//IGNORE is not supported.
7|Crawler  |     at new Iconv (/var/www/makecontact.crawl/node_modules/iconv/lib/iconv.js:52:11)
7|Crawler  |     at convertIconv (/var/www/makecontact.crawl/node_modules/encoding/lib/encoding.js:76:13)
7|Crawler  |     at convert (/var/www/makecontact.crawl/node_modules/encoding/lib/encoding.js:39:22)
7|Crawler  |     at Response.Body._convert (/var/www/makecontact.crawl/node_modules/node-fetch/lib/body.js:221:9)
7|Crawler  |     at PassThrough.<anonymous> (/var/www/makecontact.crawl/node_modules/node-fetch/lib/body.js:148:17)
7|Crawler  |     at emitNone (events.js:111:20)
7|Crawler  |     at PassThrough.emit (events.js:208:7)
7|Crawler  |     at endReadableNT (_stream_readable.js:1055:12)
7|Crawler  |     at _combinedTickCallback (internal/process/next_tick.js:138:11)
7|Crawler  |     at process._tickDomainCallback (internal/process/next_tick.js:218:9)`

**Comment by bitinn** [comment]
As hinted in error this is not from node-fetch, but from `iconv`/`encoding` module, which node-fetch v1 depends on, encoding conversion is done when you call `res.text()` on non UTF-8 HTML page.

Depends on your situation you can:

- Use node-fetch v2, it no longer do any conversion to encoding on `res.text()`, per Fetch Spec, we only support UTF-8 encoding.

- Use node-fetch v2, but use `res.textConverted()`, which does basically the same conversion as `res.text()` in node-fetch v1. If you still run into this error (and can't `catch` it), provide some test code to reproduce this issue. If verified, we can consider replacing `encoding` with another library.

- Use node-fetch v2, but use `res.body` and do you own encoding conversion if dealing with non UTF-8 content. It's basically a node stream so you have full control.

TL;DR: don't use node-fetch v1, even if we can reproduce this error consistently, it's quite like we won't have time to fix it.

**Comment by makecontact** [comment]
Thank you very much for your response.
